packages/webapp/components/invite/Referral.tsx

```diff
@@ -0,0 +1,221 @@
+import { useToastNotification } from '@dailydotdev/shared/src/hooks/useToastNotification';
+import { useAnalyticsContext } from '@dailydotdev/shared/src/contexts/AnalyticsContext';
+import { AnalyticsEvent } from '@dailydotdev/shared/src/lib/analytics';
+import { ReferralCampaignKey } from '@dailydotdev/shared/src/hooks';
+import { downloadBrowserExtension } from '@dailydotdev/shared/src/lib/constants';
+import { FlexCentered } from '@dailydotdev/shared/src/components/utilities';
+import BrowsersIcon from '@dailydotdev/shared/icons/browsers.svg';
+import { mobileL, laptopL } from '@dailydotdev/shared/src/styles/media';
+import useMedia from '@dailydotdev/shared/src/hooks/useMedia';
+import { Modal } from '@dailydotdev/shared/src/components/modals/common/Modal';
+import {
+  ModalKind,
+  ModalSize,
+} from '@dailydotdev/shared/src/components/modals/common/types';
+import CloseButton from '@dailydotdev/shared/src/components/CloseButton';
+import { JoinPageProps } from './common';
+
+export function Referral({
+  referringUser,
+  redirectTo,
+  token,
+}: JoinPageProps): ReactElement {
+  const router = useRouter();
+  const genericReferral = useFeature(feature.genericReferral);
+  const { completeAction } = useActions();
+  const { trackEvent } = useAnalyticsContext();
+  const { displayToast } = useToastNotification();
+  const { user, refetchBoot, showLogin } = useAuthContext();
+  const isMobile = !useMedia([mobileL.replace('@media ', '')], [true], false);
+  const isLaptopL = useMedia([laptopL.replace('@media ', '')], [true], false);
+  const {
+    mutateAsync: onAcceptMutation,
+    isLoading,
+    isSuccess,
+  } = useMutation(acceptFeatureInvitation, {
+    onSuccess: async () => {
+      await Promise.all([
+        completeAction(ActionType.AcceptedGenericReferral),
+        refetchBoot(),
+      ]);
+      router.push(redirectTo);
+    },
+    onError: (err: ApiErrorResult) => {
+      const message = err?.response?.errors?.[0]?.message;
+      displayToast(message ?? DEFAULT_ERROR);
+    },
+  });
+  const [isVideoOpen, setVideoOpen] = useState(false);
+
+  const onVideoModalClose = () => {
+    setVideoOpen(false);
+  };
+
+  const handleAcceptClick = () => {
+    const handleAccept = () =>
+      onAcceptMutation({
+        token,
+        referrerId: referringUser.id,
+        feature: ReferralCampaignKey.Generic,
+      });
+
+    if (!user) {
+      return showLogin({
+        trigger: AuthTriggers.GenericReferral,
+        options: {
+          onLoginSuccess: handleAccept,
+          onRegistrationSuccess: handleAccept,
+        },
+      });
+    }
+
+    // since in the page view, query params are tracked automatically,
+    // we don't need to send the params here explicitly
+    trackEvent({ event_name: AnalyticsEvent.AcceptGeneralReferralInvitation });
+
+    return handleAccept();
+  };
+
+  useEffect(() => {
+    if (genericReferral === GenericReferral.Control) {
+      return;
+    }
```

Comment:
This will depend on if it's run as experiment or not.

Reply:
Removed useEffect